Bookish.DataAccess/BookishService.cs

```diff
@@ -87,5 +90,39 @@ ORDER BY
 
             return dbConnection.Query<BookLoan>(query, new { UserID = userId });
         }
+
+        public void AddBook(string title, string author, string isbn, int totalCopies)
+        {
+            var lastId = LastCopiesId();
+
+            var valuesList = new List<string>();
+            for (var i = 1; i <= totalCopies; i++)
+            {
+                valuesList.Add($"({lastId + i}, @ISBN)");
+            }
+
+            var query =
+                @"INSERT INTO Books (ISBN, Title, Author)
+                VALUES (@ISBN, @Title, @Author); " +
+                @"INSERT INTO BookCopies (CopyId, ISBN) VALUES " +
+                string.Join(", ", valuesList);
```

Comment:
You should pass this in like a parameter 🙂

Reply:
I'm not sure I understand, can you please expand on this?

Reply:
So pass it in like you're pasing in Title, Author etc., by passing it to the execute command, instead of manually appending to the string. See https://www.learndapper.com/parameters for information about dapper parameters. It's good to use for consistency and because it protects against SQL injection. Does that make sense? 🙂

Reply:
I believe I should be using parameters since each valuesList entry should look like "(12, @ISBN)" where ISBN is passed using parameters.
I'm currently changing the CopyID such that it auto-increments meaning it would just be "(@ISBN)".

Reply:
The current version should also be safe since the CopyID isn't affected by user input.

Reply:
Yes, that's true. So what I meant was passing in the string of "(${isbn})" repeats as the parameter, instead of just isbn. That's more for consistency rather than safety. Because currently you're passing in a parameter and manually concatinating the string which is a bit confusing. But that might be a slightly non-standard way to use a parameter actually... So feel free to do it your way using Enumerable.Repeat and I'll have a look at that. Hope that isn't too confusing! I've got an interview now but happy to explain further after if that would be useful 🙂